File: HackerRank/Reto-dia-25/main.js
function isPrime(n) {
    if (n <= 1) return false;
    if (n <= 3) return true;
    if (n % 2 === 0 || n % 3 === 0) return false;
    
    for (let i = 5; i * i <= n; i += 6) {
        if (n % i === 0 || n % (i + 2) === 0) return false;
    }
    return true;
}

function main() {
    const fs = require('fs');
    const input = fs.readFileSync('/dev/stdin', 'utf-8').trim().split(/\s+/);
    
    if (input.length === 0 || input[0] === '') return;
    
    const t = parseInt(input[0], 10);
    for (let i = 1; i <= t; i++) {
        const n = parseInt(input[i], 10);
        if (isPrime(n)) {
            console.log("Prime");
        } else {
            console.log("Not prime");
        }
    }
}

main();

File: HackerRank/Reto-dia-25/main.cs
using System;

class Solution {
    static bool IsPrime(int n) {
        if (n <= 1) return false;
        if (n <= 3) return true;
        if (n % 2 == 0 || n % 3 == 0) return false;
        
        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0) return false;
        }
        return true;
    }

    static void Main(string[] args) {
        string input = Console.ReadLine();
        if (string.IsNullOrEmpty(input)) return;
        
        int t = int.Parse(input.Trim());
        for (int i = 0; i < t; i++) {
            int n = int.Parse(Console.ReadLine().Trim());
            if (IsPrime(n)) {
                Console.WriteLine("Prime");
            } else {
                Console.WriteLine("Not prime");
            }
        }
    }
}